subgraphs/compound-v3/src/sdk/account.ts

```diff
@@ -0,0 +1,331 @@
+      _account.positionCount = INT_ZERO;
+      _account.openPositionCount = INT_ZERO;
+      _account.closedPositionCount = INT_ZERO;
+      _account.depositCount = INT_ZERO;
+      _account.withdrawCount = INT_ZERO;
+      _account.borrowCount = INT_ZERO;
+      _account.repayCount = INT_ZERO;
+      _account.liquidateCount = INT_ZERO;
+      _account.liquidationCount = INT_ZERO;
+      _account.transferredCount = INT_ZERO;
+      _account.receivedCount = INT_ZERO;
+      _account.flashloanCount = INT_ZERO;
+      _account.save();
+      this.isNew = true;
+    } else {
+      this.isNew = false;
+    }
+    this.account = _account;
+  }
+
+  // returns true if the account was created in this instance
+  isNewAccount(): boolean {
+    return this.isNew;
+  }
+
+  addPosition(
+    asset: Bytes,
+    newBalance: BigInt,
+    side: string,
+    transactionType: string,
+    priceUSD: BigDecimal,
+    interestType: string | null = null
+  ): void {
+    let counterID = this.account.id
+      .toHexString()
+      .concat("-")
+      .concat(this.market.id.toHexString())
+      .concat("-")
+      .concat(side);
+    if (interestType) {
+      counterID = counterID.concat("-").concat(interestType);
+    }
+    let positionCounter = PositionCounter.load(counterID);
+    if (!positionCounter) {
+      positionCounter = new PositionCounter(counterID);
+      positionCounter.nextCount = 0;
+      positionCounter.save();
+    }
+    const positionID = positionCounter.id
+      .concat("-")
+      .concat(positionCounter.nextCount.toString());
+
+    let position = Position.load(positionID);
+    const openPosition = position == null;
+    if (openPosition) {
+      position = new Position(positionID);
+      position.account = this.account.id;
+      position.market = this.market.id;
+      position.asset = asset;
+      position.hashOpened = this.event.transaction.hash;
+      position.blockNumberOpened = this.event.block.number;
+      position.timestampOpened = this.event.block.timestamp;
+      position.side = side;
+      if (interestType) {
+        position.type = interestType;
+      }
+      position.balance = BIGINT_ZERO;
+      position.depositCount = INT_ZERO;
+      position.withdrawCount = INT_ZERO;
+      position.borrowCount = INT_ZERO;
+      position.repayCount = INT_ZERO;
+      position.liquidationCount = INT_ZERO;
+      position.transferredCount = INT_ZERO;
+      position.receivedCount = INT_ZERO;
+      position.save();
+    }
+    position = position!;
+    position.balance = newBalance;
+    if (transactionType == TransactionType.DEPOSIT) {
+      position.depositCount += INT_ONE;
+    } else if (transactionType == TransactionType.BORROW) {
+      position.borrowCount += INT_ONE;
+    } else if (transactionType == TransactionType.TRANSFER) {
+      position.receivedCount += INT_ONE;
+    }
+    // Note: liquidateCount is not incremented here
+    position.save();
+
+    if (openPosition) {
```

Comment:
You like these duplicated conditionals huh 😅 

I think you can refactor it into:
```
if (!openPosition) {
  updatePositionCounter(transactionType);
  takeSnapshot();
  return;
}

// create position and updatePositionCounter() on it
// update account open position counters
// update market open position counters
// update protocol open position counters
// takeSnapshot()

Reply:
Nice, I changed that